lib/cfn-nag/custom_rules/KMSKeyWildcardPrincipalRule.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class KMSKeyWildcardPrincipalRule < BaseRule
+  def rule_text
+    'KMS key should not allow * principal'
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F76'
+  end
+
+  def audit_impl(cfn_model)
+    # Select all AWS::KMS::Key resources to audit
+    violating_keys = cfn_model.resources_by_type('AWS::KMS::Key').select do |key|
+      # Select all key policy 'Statement' objects to audit
+      violating_statements = key.keyPolicy['Statement'].select do |statement|
+        # Add statement as violating if allowing wildcard principal
+        (statement['Principal'] == '*' || statement['Principal']['AWS'] == '*') && statement['Effect'] == 'Allow'
```

Comment:
i was just thinking about this.... we may need to enhanced the logic around SQS and SNS.  i believe the following is legal syntax:

"AWS": [
  "aws:arn:us-east-1:111111111:user/foo",
  "*"
]

Reply:
Yeah, we may need to make use of the parsing in cfn-model for principals.  I believe that parsing (and the `wildcard_principal` method) take the depth into account.  Let's hold off merging this, and I'll start reworking this to use that model.
https://github.com/stelligent/cfn-model/blob/master/lib/cfn-model/model/principal.rb